### PR TITLE
ZMQ video streaming + record/replay functionality update

### PR DESCRIPTION
**Pull request by crazyhathello**
- Implement video streaming node for ZMQ streaming of head video
- Updated record node to support start/stop of recordings on the fly
- Added replay script to write trajectories to shared memory

### CODE REVIEW

**Comment by ProKil** [comment]
Can you move the dataflow files into dataflow/examples?

File: dataflows/stretch_zmq_streaming.toml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+height_resize_factor = 0.1
+
+[[nodes]]
+node_name = "video_streaming"
+node_class = "head_video_streamer"
+
+[nodes.node_args]
+input_channel = "meta_2_head_cam"
+output_channel = "head_video_status"
+quest_controller_ip = "172.26.172.110"
```
Comment (ProKil) [inline dataflows/stretch_zmq_streaming.toml:40]: Maybe change it to <your_quest_ip>

**Review by ProKil — CHANGES_REQUESTED** [review]

File: teleop/vr_controller_node.py [hunk 2]
```diff
@@ -187,24 +186,30 @@ def _convert_controller_states_to_control_signals(
...
-                json.dumps(
-                    dict(
-                        controller_states=controller_states.model_dump(),
-                        target_position=target_position.model_dump(),
-                    )
-                )
-                + "\n"
-            )
+            target_position.record_button = True
+            # if self.recording_file is None:
```
[the remark this answers is not in the document]
Comment (sophiameerkat) [reply teleop/vr_controller_node.py]: We have removed the commented lines in the following commit

File: dataflows/examples/stretch_zmq_streaming.toml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+[[nodes]]
+node_name = "video_streaming"
+node_class = "head_video_streamer"
+
+[nodes.node_args]
+input_channel = "meta_2_head_cam"
+output_channel = "head_video_status"
+quest_controller_ip = "172.26.172.110"
+
+# [[nodes]]
```
[the remark this answers is not in the document]
Comment (sophiameerkat) [reply dataflows/examples/stretch_zmq_streaming.toml]: We have removed the commented lines and update the redis and quest ip to placeholder in the following commit

File: dataflows/stretch_zmq_streaming.toml (hunk 1 above)
Comment (sophiameerkat) [reply dataflows/stretch_zmq_streaming.toml]: We have removed the commented lines and update the redis and quest ip to placeholder in the following commit

File: dataflows/examples/quest_local_redis.toml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+[[nodes]]
+node_name = "quest_3_controller"
+node_class = "quest_controller"
+
+[nodes.node_args]
+input_channel = "stretch_status"
+output_channel = "quest_control"
+translation_speed = 20
+quest_controller_ip = "172.26.172.110"
```
Comment (ProKil) [inline dataflows/examples/quest_local_redis.toml:13]: maybe change this as well?

**Review by ProKil — CHANGES_REQUESTED** [review]

**Comment by ProKil** [comment]
@sophiameerkat @crazyhathello Thanks! Could you update this last change and I will approve it.

**Review by ProKil — APPROVED** [review]

**Pull request merged**